pytorch_lightning/utilities/auto_restart.py

```diff
@@ -70,31 +70,37 @@ def worker_id(self) -> int:
 
     def __iter__(self) -> Iterator[Any]:
         self._current_iteration = 0
+        self.i = 0
+        self.sampler_iter = iter(self._sampler)
```

Comment:
these attributes are undefined in the constructor. `i` is not an expressive name for a public variable

Reply:
Done !